crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -95,19 +130,31 @@ impl Inner {
                     // This should happen only for prices missing while building the auction.
                     // Otherwise malicious actors could easily cause the cache size to blow up.
                     let outdated_timestamp = now.checked_sub(*max_age).unwrap();
-                    entry.insert(CachedResult {
-                        result: Ok(0.),
-                        updated_at: outdated_timestamp,
-                        requested_at: now,
-                    });
+                    entry.insert(CachedResult::new(
+                        Ok(0.),
+                        outdated_timestamp,
+                        now,
+                        Default::default(),
+                    ));
                 }
                 None
             }
         }
     }
 
+    fn get_ready_to_use_cached_price(
```

Comment:
This function doesn't feel like it's carrying its own weight. I think inlining it in the 2 places where it's used is fine and similarly expressive.

Reply:
I just wanted to hide the filtering details there and avoid using a non-filtered cache anywhere else.